bin/extract_api_surface.py

```
@@ -0,0 +1,207 @@
+        return f"[{', '.join(items)}]"
+    if isinstance(node, ast.Tuple):
+        items = [str(_default_to_str(e) or "") for e in node.elts]
+        return f"({', '.join(items)})"
+    if isinstance(node, ast.UnaryOp) and isinstance(node.op, ast.USub):
+        return f"-{_default_to_str(node.operand)}"
+    if isinstance(node, ast.Call):
+        return f"{_annotation_to_str(node.func)}(...)"
+    return "..."
+
+
+def _signature_from_function(func_node):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_signature_from_function`

Add return type annotation: `str`

(ANN202)